x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/step_define_rule/index.tsx

```diff
@@ -154,64 +164,76 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
     setOpenTimelineSearch(false);
   }, []);
 
-  return isReadOnlyView && myStepData?.queryBar != null ? (
+  return isReadOnlyView ? (
     <StepContentWrapper addPadding={addPadding}>
       <StepRuleDescription
         direction={descriptionDirection}
         indexPatterns={indexPatternQueryBar as IIndexPattern}
-        schema={schema}
-        data={myStepData}
+        // @ts-ignore schema is insufficiently typed, but this works
```

Comment:
We want to avoid all `@ts-ignore` as it should never be needed anywhere in the codebase except for within the import statements when we are pulling in JavaScript code unless we notice for some reason a new pattern that can not be handled with either:

* regular type script changes
* type narrowing through io-ts or "is casting functions"
* as cast
* If all else fails a @ts-ignore and a team conversation about why it is there.

And manage it in that order best to least best. Unit tests are kind of "ok" when we are pinched for time since that is adding more testing and testing is the most noble cause our team can contribute next to features and bug fixes ;-)

For this one I was able to fix like so using type script changes:
<img width="2330" alt="Screen Shot 2020-03-17 at 11 29 19 AM" src="https://user-images.githubusercontent.com/1151048/76884304-202ae100-6843-11ea-909b-33e9ea7b1572.png">

I changed these types:

```ts
interface RuleFields {
  anomalyThreshold?: unknown;
  mlJobId?: unknown;
  queryBar?: unknown;
  index?: unknown;
  ruleType?: unknown;
}
```

And then for extra fun I use your same pattern below to get more type safety and remove lodash omit as well:
```ts
export const filterRuleFieldsForType = <T extends RuleFields>(fields: T, type: RuleType) => {
  if (isMlRule(type)) {
    const { index, queryBar, ...withoutIndexQueryBar } = fields;
    return withoutIndexQueryBar;
  } else {
    const { anomalyThreshold, mlJobId, ...withoutAnomalyThresholdMlJob } = fields;
    return withoutAnomalyThresholdMlJob;
  }
};
```

You did a great job of adding that function which made it a joy for me to refactor into stronger types. The two pros of trying to avoid lodash.omit when possible are:

* lodash omit is not as type safe with the string array and we have to hunt for strings when we change variable names or have spelling mistakes
* omit is actually deprecated from lodash, https://github.com/lodash/lodash/wiki/Roadmap, https://github.com/lodash/lodash/issues/2930

Reply:
So as per my `ts-ignore` comment there, the issue here is not with the helpers but with the schema's `formData` interface. making `RuleFields` totally optional fixes this because you can now cast any object as RuleFields, meaning you can pass any object to that filtering helper.

However, both our data and the schema _do_ (types or no) have those fields present, and RuleFields should explicitly _not_ be optional.